src/types/types.ts

```diff
@@ -169,21 +170,61 @@ class PDAConfigurationMemento implements IConfigurationMemento {
     }
 }
 
-export interface ISimulation {
+export interface ISimulation{
     automaton: IAutomaton;
     configuration: IAutomatonConfiguration;
 
-    commandHistory: RunCommand<unknown>[];
-    executeCommand<T>(command: RunCommand<T>): void; // if (command.execute()) { commandHistory.push(command); }
+    commandHistory: RunCommand[];
+    executeCommand(command: RunCommand): void; // if (command.execute()) { commandHistory.push(command); }
     undo(): void; // command = commandHistory.pop(); command.undo();
 
     run(): void;
 }
 
-export abstract class RunCommand<T = void> {
+export class Simulation implements ISimulation {
+    automaton: IAutomaton;
+    configuration: IAutomatonConfiguration;
+    commandHistory: (RunCommand)[];
+
+    constructor(_automaton: IAutomaton, _configuration: IAutomatonConfiguration){
+            this.automaton = _automaton;
+            this.configuration = _configuration;
+            this.commandHistory = [];
+    }
+
+    executeCommand(command: RunCommand): void {
+        if (command.execute()) {
+             this.commandHistory.push(command); 
+            }
+        // maybe return error if fails to execute
+    }
+    undo(): void {
+        const command = this.commandHistory.pop();
+        if (command === undefined) return; // maybe error message when we try undo on empty history
+            else command.undo();
```

Comment:
```suggestion
        if (command === undefined) {
            return; // maybe error message when we try undo on empty history
        } else {
            command.undo();
        }
```

Please use { and } even if one-liner blocks for safety.